### DISCUSSION TITLE
Is there a way to require a particular Node version for a plugin?

### DISCUSSION THREAD

**Discussion by hoelzro** [TiddlyWiki/TiddlyWiki5#5359] [General]
I recently shared a new version of a plugin I wrote with a friend, and he mentioned it didn't work for him - it turns out my changes to the code made use of some JavaScript syntax present in Node 15 but not Node 14.  I know one can specify a required version of TiddlyWiki itself via `core-version` (but unless I'm missing something, this doesn't seem to be enforced) - is there something analogous for the Node version?

**Comment by pmario** [comment]
`core-version` means TW version .. not node

**Comment by Jermolene** [comment]
Hi @hoelzro the core does (did?) check the Node version against `engines.node` in `package.json` (currently still set to `>=0.8.2`!), but I don't think it would help here. I guess the cleanest current solution would be for the plugin itself to raise an error if it detects that it's running on unsupported version of Node.

The core is generally very conservative about only using ECMAScript 5/6 era Node.js features but I appreciate that's not a viable solution for all plugins.

**Comment by hoelzro** [comment]
@Jermolene Thanks, that's kind of what I figured but wanted to confirm.  I understand that core plugins are very conservative - I just like making use of newer language features for plugins I don't distribute widely!
